lib/rom/processor/transproc.rb

```diff
@@ -99,6 +99,8 @@ def visit(attribute)
       def visit_attribute(attribute)
         if attribute.typed?
           t(:map_value, attribute.name, t(:"to_#{attribute.type}"))
+        elsif attribute.meta[:coercer]
+          t(:map_value, attribute.name, attribute.meta[:coercer])
         end
```

Comment:
```
diff --git a/lib/rom/processor/transproc.rb b/lib/rom/processor/transproc.rb
index 6dbd17d..9b6d35f 100644
--- a/lib/rom/processor/transproc.rb
+++ b/lib/rom/processor/transproc.rb
@@ -91,16 +91,21 @@ module ROM
 
       # Visit plain attribute
       #
+      # It will call block transformation if its used
+      #
       # If it's a typed attribute a coercion transformation is added
       #
+      # It will raise error if block and type used in the same time
+      #
       # @param [Header::Attribute] attribute
       #
       # @api private
       def visit_attribute(attribute)
-        if attribute.typed?
-          t(:map_value, attribute.name, t(:"to_#{attribute.type}"))
-        elsif attribute.meta[:coercer]
+        raise StandardError if attribute.meta[:coercer] && attribute.typed?
+        if attribute.meta[:coercer]
           t(:map_value, attribute.name, attribute.meta[:coercer])
+        elsif attribute.typed?
+          t(:map_value, attribute.name, t(:"to_#{attribute.type}"))
         end
       end
 ```